src/json/pull_parser.cr

```diff
@@ -138,6 +210,21 @@ class JSON::PullParser
     end
   end
 
+  # Read the next value and returns it.
+  #
+  # If the value is a null, boolean, integer or string, it is returned as is.
+  # If the value is an array or an object, it returns a string representing it.
+  # If the value in unknown, it raises a `ParseException`.
+  #
+  # ```
+  # pull = JSON::PullParser.new(input)
```

Comment:
I'd suggest parsing an array of differently-typed values as a self-contained example of all read_raw results:
```cr
pull = JSON::PullParser.new %([nil, true, 1, "foo", [1, "two"], {"foo": "bar"}])
pull.read_begin_array
pull.read_raw # => nil
pull.read_raw # => true
pull.read_raw # => 1
pull.read_raw # => "foo"
pull.read_raw # => "[1, \"two\"]"
pull.read_raw # => "{\"foo\": \"bar\"}"
pull.read_end_array
```

Reply:
The values are actually all of type `String`.